src/main/java/com/squareup/javapoet/AnnotationSpec.java

```diff
@@ -235,7 +235,8 @@ private Builder addValue(String memberName, Object value) {
         return addMember(memberName, "$Lf", value);
       }
       if (value instanceof Character) {
-        return addMember(memberName, "'$L'", value);
+        String format = Character.valueOf('\'').equals(value) ? "'\\\''" : "'$L'";
```

Comment:
Tried it without and the test fails, because we really need the `'\''` defined by `char g() default '\'';`